src/inmanta/ast/type.py

```diff
@@ -334,7 +334,8 @@ def validate(self, value: Optional[object]) -> bool:
         """
         if isinstance(value, AnyType):
             return True
-
+        if references.is_reference_of(value, float):
+            return True
```

Comment:
This is the main open issue I think

Reply:
Do I understand correctly that the issue is that we don't always want to allow references, so this is too accepting?

We could have a `Reference(value_type=Type)` type, who's validate checks exactly this. Then validate callers could always be explicit in whether they want to allow references? I'd have to double check which callers exist, but I expect that for each of them being explicit would be both feasible and more clear.

Or perhaps a `validate_reference` method on `Type`. Would have the advantage that we can rely on polymorphism for the `Float -> float` mapping etc. But it's less flexible in the sense that adding support for more special cases (`Unknown` etc) requires a special method for each.